consensus/e2e_tests/utils_test.go

```diff
@@ -840,6 +840,8 @@ func advanceTime(t *testing.T, clck *clock.Mock, duration time.Duration) {
 	clck.Add(duration)
 	t.Logf("[⌚ CLOCK ⏩] advanced by %v", duration)
 	logTime(t, clck)
+	// Give goroutines a chance to run
+	clck.Add(0)
```

Comment:
This clock mock implementation seems to advance time synchronously not giving a chance to go routines to trigger. This seems to do the trick though, I can get that back to paceMaker tests but it allowed to have exact time precision.

Reply:
Is there any documentation explaining this in their godocs you could link to?

Reply:
Looking into the clock modules code, it seems that `clck.Add(duration)` executes as following:
`clck.Add->time.Sleep(1 * time.Millisecond)`. This may be easily tested, otherwise it may be happening somewhere here [`benbjohnson/clock/clock.go#L101C7-L101C7`](https://github.com/benbjohnson/clock/blob/master/clock.go#L101C7-L101C7)
```
// Continue to execute timers until there are no more before the new time.
for {
	if !m.runNextTimer(t) {
		break
	}
}
```